framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -191,6 +191,76 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
                 " is consistently oriented with regards to the blocks it neighbors";
 
     diagnosticsLog(message, _check_sidesets_orientation, flipped_pairs.size());
+
+    // Now check that there is no sideset radically flipping from one side's normal to another
+    // side next to it, in the same sideset
+    // We'll consider pi / 2 to be the most steep angle we'll pass
+    unsigned int num_normals_flipping = 0;
+    Real steepest_side_angles = 1;
```

Comment:
Why 57 degrees for the initialization of this value?

Reply:
this isnt cos(angle) it s acos(something) so 1 the upper limit